### DISCUSSION TITLE
Integration of a FOSS voice assistant

### DISCUSSION THREAD

**Discussion by FabulousOne** [AnySoftKeyboard/AnySoftKeyboard#3630] [Ideas]
Hi there,
Would it be possible to integrate [Sayboard](https://github.com/ElishaAz/Sayboard) to ASK?  I know another keyboard where it works, but I don't like this keyboard. So I'd like to see this FOSS TTS integrated in ASK. What do you think?

**Comment by menny** [comment]
That's nice! I will take a look. Thanks

**Reply by Redsandro** [reply]
Another interesting alternative is OpenAI [Whisper](https://github.com/openai/whisper) Someone [ported the "tiny" model](https://github.com/usefulsensors/openai-whisper/tree/main#readme) to Android (test app available) and someone else [made a voice keyboard](https://github.com/MichaelMcCulloch/WhisperVoiceKeyboard). Quite alpha though, but I'm surprised by the accuracy of the demo app for such a small voice model.

Whisper was trained on a large amount of multilingual data, which could potentially give it an edge in terms of accuracy and performance.
